attribute_set/models/attribute_attribute.py

```diff
@@ -254,37 +271,36 @@ def button_add_options(self):
     def create(self, vals):
         """ Create an attribute.attribute
 
-        When a `field_id` is given, the attribute will be linked to the
-        existing field. The use case is to create an attribute on a field
-        created with Python `fields`.
+        - In case of a new "custom" attribute, a new field object 'ir.model.fields' will
+        be created as this model "_inherits" 'ir.model.fields'.
+        So we need to add here the mandatory 'ir.model.fields' instance's attributes to
+        the new 'attribute.attribute'.
+
+        - In case of a new "native" attribute, it will be linked to an existing
+        field object 'ir.model.fields' (through "field_id") that cannot be modified.
+        That's why we remove all the 'ir.model.fields' instance's attributes values
+        from `vals` before creating our new 'attribute.attribute'.
 
         """
-        if vals.get("field_id"):
-            # When a 'field_id' is given, we create an attribute on an
-            # existing 'ir.model.fields'.  As this model `_inherits`
-            # 'ir.model.fields', calling `create()` with a `field_id`
-            # will call `write` in `ir.model.fields`.
-            # When the existing field is not a 'manual' field, we are
-            # not allowed to write on it. So we call `create()` without
-            # changing the fields values.
-            field_obj = self.env["ir.model.fields"]
-            field = field_obj.browse(vals["field_id"])
+        if vals.get("nature") == "native":
 
+            field_obj = self.env["ir.model.fields"]
             if vals.get("serialized"):
                 raise ValidationError(
                     _("Error"),
                     _(
                         "Can't create a serialized attribute on "
                         "an existing ir.model.fields (%s)"
                     )
-                    % field.name,
+                    % field_obj.browse(vals["field_id"]).name,
                 )
 
-            if field.state != "manual":
-                # The ir.model.fields already exists and we want to map
-                # an attribute on it. We can't change the field so we
-                # won't add the ttype, relation and so on.
-                return super(AttributeAttribute, self).create(vals)
+            # Remove all the values that can modify the related native field
+            # before creating the new 'attribute.attribute'
+            for key in list(vals.keys()):
+                if key in field_obj.fields_get().keys():
+                    del vals[key]
```

Comment:
the code is complex and you call field_get a lot of time

```suggestion
            for key in set(vals).intersection(field_obj._fields):
                del vals[key]
```